skyportal/models/gcn.py

```diff
@@ -128,13 +139,133 @@ class GcnPublication(Base):
 
     publication_name = sa.Column(sa.String, nullable=False)
 
-    publish = sa.Column(
+    published = sa.Column(
         sa.Boolean,
         nullable=False,
         server_default='false',
         doc='Whether GcnPublication should be published',
     )
 
+    def publish(self):
+        """Publish GcnPublication."""
+        self.published = True
+        return self.generate_html()
+
+    def generate_html(self):
+        """Publish GcnPublication."""
+        # TODO: create the hmtl and cache it
+        cache_key = f"gcn_{self.id}"
+        data = self.data
+        if isinstance(data, str):
+            data = json.loads(data)
+        # TODO: use a proper template (like with jinja)
+        html = "<!DOCTYPE html>"
+        html += "<html>"
+
+        html += "<body>"
+        html += "<h1>GCN Publication</h1>"
+        html += "<br>"
+        html += "<div style='display: flex; flex-direction: row; justify-content: space-between; align-items: center;'><h2>Sources</h2>"
+        if len(data.get("sources", [])) == 0:
+            html += "</div><p>No sources</p>"
+        else:
+            html += "<button type='button' onclick='tableToCSV(\"sources\")' style='max-height: 30px;'>Download CSV</button></div>"
+            html += "<table style='width:100%' border='1' align='center' id='sources'>"
+            html += "<tr><th>Source ID</th><th>RA</th><th>Dec</th><th>Redshift</th><th>Comment</th></tr>"
+            for source in data["sources"]:
+                html += f"<tr><td>{source['id']}</td><td>{source['ra']}</td><td>{source['dec']}</td><td>{source['redshift']}</td><td>{source.get('comment', '')}</td></tr>"
+            html += "</table>"
+            html += "<br>"
+            html += "<div style='display: flex; flex-direction: row; justify-content: space-between; align-items: center;'><h2>Observations</h2>"
+
+        if len(data.get("observations", [])) == 0:
+            html += "</div><p>No observations</p>"
+        else:
+            html += "<button type='button' onclick='tableToCSV(\"observations\")' style='max-height: 30px;'>download as CSV</button></div>"
+            html += (
+                "<table style='width:100%' border='1' align='center' id='observations'>"
+            )
+            html += "<tr><th>Observation ID</th><th>Time</th><th>Instrument ID</th><th>Field ID</th><th>Filter</th><th>Exp. Time</th><th>Limiting Mag.</th><th>Airmass%</th><th>Processed %</th></tr>"
```

Comment:
@Theodlz Super minor, but this is not technically a %, just a fraction. Could multiply by 100 to get to %.